server/api/team/team.controller.js

```diff
@@ -73,6 +74,66 @@ exports.players = function(req, res) {
     });
 };
 
+exports.fill = function(req, res) {
+    var currentSport = ""
+    var user = req.user;
+      if (user.role !== admin) {
+        logger.log("error", "Filling teams is restricted to admins only");
+        return res.send (403);
+      }
+    Sport.find(req.params.id).then(function(sport) {
+        currentSport = sport
+    }).then(function() {
+        Team.findAll({
+            where: [{special: null, sportId: currentSport.id}]
```

Comment:
I'm not sure Express is quite smart enough to parse the URL into separate objects. I think it ends up looking like /leagues/:league_id/sport/:sport_id/teams and then you get req.params.league_id and so on.
I don't think you need to add a query to find a league -- Just add "LeagueId: req.params.league_id" to the Team where clause.